plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -0,0 +1,337 @@
+					target: 'businessInfo',
+					cond: () => true, // TODO: use a custom function to check on the parallel state using meta when we implement that. https://xstate.js.org/docs/guides/guards.html#guards-condition-functions
+				},
+			],
+			meta: {
+				progress: 50,
+			},
+		},
+		businessInfo: {
+			on: {
+				BUSINESS_INFO_COMPLETED: {
+					target: 'preExtensions',
+					actions: [
+						assign( {
+							businessInfo: (
+								_context,
+								event: BusinessInfoEvent
+							) => event.payload.businessInfo, // assign context.businessInfo to the payload of the event
+						} ),
+					],
+				},
+			},
+			meta: {
+				progress: 60,
+				component: BusinessInfo,
+			},
+		},
+		skipFlowBusinessLocation: {
+			on: {
+				BUSINESS_LOCATION_COMPLETED: {
+					target: 'settingUpStore',
+					actions: [
+						assign( {
+							businessInfo: (
+								_context,
+								event: BusinessLocationEvent
+							) => event.payload.businessInfo, // assign context.businessInfo to the payload of the event
+						} ),
+					],
+				},
+			},
+			meta: {
+				progress: 80,
+				component: BusinessLocation,
+			},
+		},
+		preExtensions: {
+			always: [
+				// immediately transition to extensions without any events as long as extensions fetching parallel has completed
+				{
+					target: 'extensions',
+					cond: () => true, // TODO: use a custom function to check on the parallel state using meta when we implement that. https://xstate.js.org/docs/guides/guards.html#guards-condition-functions
+				},
+			],
+			// add exit action to filter the extensions using a custom function here and assign it to context.extensionsAvailable
+			exit: assign( {
+				extensionsAvailable: ( context ) => {
+					return context.extensionsAvailable.filter( () => true );
+				}, // TODO : define an extensible filter function here
+			} ),
+			meta: {
+				progress: 70,
+			},
+		},
+		extensions: {
+			on: {
+				EXTENSIONS_COMPLETED: {
+					target: 'settingUpStore',
+				},
+			},
+			meta: {
+				progress: 80,
+				component: Extensions,
+			},
+		},
+		settingUpStore: {},
+	},
+} );
+
+const CoreProfilerController = ( {} ) => {
+	const [ state, send ] = useMachine(
+		coreProfilerStateMachineDefinition.withConfig( {
+			services: {
+				initializeProfiler: () => ( sendBack ) => {
+					const interval = setInterval( () => {
+						sendBack( {
+							type: 'INITIALIZATION_COMPLETE',
+						} );
+					}, 1000 );
+
+					return () => {
+						clearInterval( interval );
+					};
+				},
+			},
+		} ),
+		{ devTools: true }
+	);
+
+	const currentNodeMeta = state.meta[ `coreProfiler.${ state.value }` ]
+		? state.meta[ `coreProfiler.${ state.value }` ]
+		: undefined;
+	const navigationProgress = currentNodeMeta?.progress; // This value is defined in each state node's meta tag, we can assume it is 0-100
+	const CurrentComponent =
+		currentNodeMeta?.component ?? ( () => <div>Insert Spinner</div> ); // If no component is defined for the state then its a loading state
```

Comment:
How about using an actual spinner component here?

Reply:
I was leaving the aesthetics to a secondary issue but I forgot to mention that in the follow up tasks! 

I'll add that, since I think it involves more than just changing this component to a spinner.